src/lava/proc/io/bridge/extractor.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+import typing as ty
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import InPort
+
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.model.py.ports import PyOutPort, PyInPort
+
+from lava.magma.compiler.channels.pypychannel import PyPyChannel
+from lava.magma.runtime.message_infrastructure.multiprocessing import \
+    MultiProcessing
+
+from lava.proc.io.bridge.utils import ChannelConfig, ChannelSendBufferFull, \
+    ChannelRecvBufferEmpty, ChannelRecvBufferNotEmpty,\
+    validate_shape, validate_buffer_size, validate_channel_config, \
+    send_data_blocking, send_data_non_blocking_drop, \
+    recv_empty_blocking, recv_empty_non_blocking_zeros, recv_not_empty_fifo, \
+    recv_not_empty_accumulate
+
+
+class Extractor(AbstractProcess):
+    """Process that gets data out of Lava.
+
+    The Extractor Process exposes a method, recv_data, which enables users to
+    receive data in external Python scripts from the ProcessModel of the
```

Comment:
Think about this from the perspective of What problem does this process solve?

E.g. "Extractor allows non-Lava code, such as a third-party Python library or an actuator, to extract data from a synchronous Lava process by calling recv_data once per timestep."